cfme/infrastructure/deployment_roles.py

```diff
@@ -1,67 +1,265 @@
 """ A model of an Infrastructure Deployment roles in CFME"""
 
+from cfme.base.ui import BaseLoggedInPage
+from cfme.exceptions import ItemNotFound, RoleNotFound
+from cfme.infrastructure.provider.openstack_infra import OpenstackInfraProvider
 from functools import partial
 from navmazing import NavigateToAttribute, NavigateToSibling
-from selenium.common.exceptions import NoSuchElementException
-
-from cfme.exceptions import ListAccordionLinkNotFound
-from cfme.fixtures import pytest_selenium as sel
-from cfme.infrastructure.provider.openstack_infra import OpenstackInfraProvider
+from utils.appliance.implementations.ui import CFMENavigateStep, navigator, navigate_to
 from utils.appliance import Navigatable
-from utils.appliance.implementations.ui import (CFMENavigateStep, navigate_to,
-                                                navigator)
-from cfme.web_ui import (listaccordion as list_acc, match_location, Quadicon,
-                         Region, toolbar as tb)
-from utils import version
-from utils.pretty import Pretty
-
-
-details_page = Region(infoblock_type='detail')
-
-cfg_btn = partial(tb.select, 'Configuration')
-pol_btn = partial(tb.select, 'Policy')
-match_page = partial(match_location, controller='ems_cluster')
-
-
-class DeploymentRoles(Pretty, Navigatable):
+from widgetastic.exceptions import NoSuchElementException
+from widgetastic.utils import Version, VersionPick
+from widgetastic.widget import ParametrizedView, View
+from widgetastic_manageiq import (
+    Accordion,
+    BaseEntitiesView,
+    BaseListEntity,
+    BaseQuadIconEntity,
+    BaseTileIconEntity,
+    BootstrapSelect,
+    BootstrapTreeview,
+    BreadCrumb,
+    ItemsToolBarViewSelector,
+    JSBaseEntity,
+    NonJSBaseEntity,
+    PaginationPane,
+    SummaryTable,
+    Table,
+    Text,
+)
+from widgetastic_patternfly import (
+    BootstrapNav,
+    Button,
+    Dropdown,
+    FlashMessages
+)
```

Comment:
**REQUIRED**
Please read style guide more carefully:

> In addition to being broken up into the three sections of standard library, third-party, and the local application, imports should be sorted alphabetically. ‘import’ lines within those sections still come before ‘from ... import’ lines:

```python
from functools import partial

from navmazing import NavigateToAttribute, NavigateToSibling
from widgetastic.exceptions import NoSuchElementException
from widgetastic.utils import Version, VersionPick
from widgetastic.widget import ParametrizedView, View
from widgetastic_manageiq import (
    Accordion,
    BaseEntitiesView,
    BaseListEntity,
    BaseQuadIconEntity,
    BaseTileIconEntity,
    BootstrapSelect,
    BootstrapTreeview,
    BreadCrumb,
    ItemsToolBarViewSelector,
    JSBaseEntity,
    NonJSBaseEntity,
    SummaryTable,
    Table,
    Text
 )
 from widgetastic_patternfly import (
     BootstrapNav,
     Button,
     Dropdown,
     FlashMessages
 )

from cfme.base.ui import BaseLoggedInPage
from cfme.exceptions import ItemNotFound, RoleNotFound
from cfme.infrastructure.provider.openstack_infra import OpenstackInfraProvider
```

Reply:
Thank you so much @quarckster , You taught me a lot in this PR.

Reply:
@digitronik FYI widgetastic_manageiq is correctly considered as an external library even though is local, because it will once become an external library.